// @shopify/cli/dist/cli/commands/demo/index.js
import {
  demo
} from "../../../chunk-G56QICQP.js";
import "../../../chunk-7P2SWIQY.js";
import {
  base_command_default
} from "../../../chunk-7VU6FOLD.js";
import "../../../chunk-NAXATNVY.js";
import "../../../chunk-LKGDG6WW.js";
import "../../../chunk-Y4AOSTJD.js";
import "../../../chunk-TMGCRPEZ.js";
import "../../../chunk-BUQTR47W.js";
import "../../../chunk-URD52LAO.js";
import "../../../chunk-2EVDXDJS.js";
import "../../../chunk-55F2DQCB.js";
import "../../../chunk-23EONB5V.js";
import {
  readFile
} from "../../../chunk-ZLSD2RNR.js";
import "../../../chunk-FQIRJADJ.js";
import "../../../chunk-FBB6KUZG.js";
import "../../../chunk-OVPEO3GL.js";
import {
  require_lib
} from "../../../chunk-4AOK4O3R.js";
import "../../../chunk-FWGRGI4R.js";
import "../../../chunk-BE73NZXB.js";
import "../../../chunk-VNBOO6TJ.js";
import "../../../chunk-QTHWTF4X.js";
import "../../../chunk-RQ24VTW4.js";
import "../../../chunk-WCNR75S2.js";
import {
  cwd,
  resolvePath
} from "../../../chunk-G6FN5VUE.js";
import "../../../chunk-3JLUTHGR.js";
import "../../../chunk-GXPKATXW.js";
import {
  __toESM,
  init_cjs_shims
} from "../../../chunk-M63RTPGR.js";

// src/cli/commands/demo/index.ts
init_cjs_shims();
var import_core = __toESM(require_lib(), 1);
var Demo = class _Demo extends base_command_default {
  static {
    this.description = "Demo a command design defined in a file";
  }
  static {
    this.hidden = true;
  }
  static {
    this.flags = {
      file: import_core.Flags.string({
        hidden: false,
        description: "The path to the design file.",
        env: "SHOPIFY_FLAG_PATH",
        parse: async (input) => resolvePath(input),
        default: async () => cwd()
      })
    };
  }
  async run() {
    const { flags } = await this.parse(_Demo);
    const contents = await readFile(flags.file);
    const design = JSON.parse(contents);
    await demo(design);
  }
};
export {
  Demo as default
};
