// @shopify/cli/dist/cli/commands/docs/generate.test.js
import {
  extractCommandData,
  writeCommandDocumentation,
  writeCommandFlagInterface,
  writeCommandUsageExampleFile
} from "../../../chunk-RNLLSL24.js";
import {
  describe,
  globalExpect,
  test,
  vi
} from "../../../chunk-5R5555YE.js";
import "../../../chunk-7VU6FOLD.js";
import "../../../chunk-NAXATNVY.js";
import "../../../chunk-LKGDG6WW.js";
import "../../../chunk-Y4AOSTJD.js";
import "../../../chunk-TMGCRPEZ.js";
import "../../../chunk-BUQTR47W.js";
import "../../../chunk-URD52LAO.js";
import "../../../chunk-2EVDXDJS.js";
import "../../../chunk-55F2DQCB.js";
import "../../../chunk-23EONB5V.js";
import {
  writeFile
} from "../../../chunk-ZLSD2RNR.js";
import "../../../chunk-FQIRJADJ.js";
import "../../../chunk-FBB6KUZG.js";
import "../../../chunk-OVPEO3GL.js";
import "../../../chunk-4AOK4O3R.js";
import "../../../chunk-FWGRGI4R.js";
import "../../../chunk-BE73NZXB.js";
import "../../../chunk-VNBOO6TJ.js";
import "../../../chunk-QTHWTF4X.js";
import "../../../chunk-RQ24VTW4.js";
import "../../../chunk-WCNR75S2.js";
import "../../../chunk-G6FN5VUE.js";
import "../../../chunk-3JLUTHGR.js";
import "../../../chunk-GXPKATXW.js";
import {
  init_cjs_shims
} from "../../../chunk-M63RTPGR.js";

// src/cli/commands/docs/generate.test.ts
init_cjs_shims();
vi.mock("@shopify/cli-kit/node/fs");
var testCommand = {
  aliases: [],
  args: {},
  flags: {
    flag1: {
      name: "flag1",
      type: "option",
      char: "f",
      description: "flag description"
    },
    flag2: {
      name: "flag2",
      type: "boolean",
      char: "a",
      description: "another flag description",
      env: "FLAG_ENV",
      allowNo: false
    }
  },
  descriptionWithMarkdown: "command markdown description",
  description: "command description",
  summary: "command summary",
  hidden: false,
  hiddenAliases: [],
  id: "topic:test-command",
  load: () => Promise.resolve({})
};
var commandData = {
  commandName: "topic test-command",
  fileName: "topic-test-command",
  interfaceName: "topictestcommand",
  hasTopic: true,
  topic: "topic",
  hasFlags: true
};
describe("extractCommandData", () => {
  test("returns the correct data", () => {
    globalExpect(extractCommandData(testCommand)).toEqual(commandData);
  });
});
describe("writeCommandFlagInterface", () => {
  test("calls writeFile with the correct content", async () => {
    await writeCommandFlagInterface(testCommand, commandData);
    globalExpect(writeFile).toHaveBeenCalledWith(
      globalExpect.stringContaining("docs-shopify.dev/commands/interfaces/topic-test-command.interface.ts"),
      `// This is an autogenerated file. Don't edit this file manually.
export interface topictestcommand {
  /**
   * flag description
   *
   */
  '-f, --flag1 <value>'?: string

  /**
   * another flag description
   * @environment FLAG_ENV
   */
  '-a, --flag2'?: ''
}
`
    );
  });
  test("calls writeFile with the correct content for a command with no flags", async () => {
    await writeCommandFlagInterface({ ...testCommand, flags: {} }, commandData);
    globalExpect(writeFile).toHaveBeenCalledWith(
      globalExpect.stringContaining("docs-shopify.dev/commands/interfaces/topic-test-command.interface.ts"),
      `// This is an autogenerated file. Don't edit this file manually.
export interface topictestcommand {

}
`
    );
  });
});
describe("writeCommandDocs", () => {
  test("calls writeFile with the correct content", async () => {
    await writeCommandDocumentation(testCommand, commandData);
    globalExpect(writeFile).toHaveBeenCalledWith(
      globalExpect.stringContaining("docs-shopify.dev/commands/topic-test-command.doc.ts"),
      `// This is an autogenerated file. Don't edit this file manually.
import {ReferenceEntityTemplateSchema} from '@shopify/generate-docs'

const data: ReferenceEntityTemplateSchema = {
  name: 'topic test-command',
  description: \`command markdown description\`,
  overviewPreviewDescription: \`command summary\`,
  type: 'command',
  isVisualComponent: false,
  defaultExample: {
    codeblock: {
      tabs: [
        {
          title: 'topic test-command',
          code: './examples/topic-test-command.example.sh',
          language: 'bash',
        },
      ],
      title: 'topic test-command',
    },
  },
  definitions: [
  {
    title: 'Flags',
    description: 'The following flags are available for the \`topic test-command\` command:',
    type: 'topictestcommand',
  },
  ],
  category: 'topic',
  related: [
  ],
}

export default data`
    );
  });
});
describe("writeCommandUsageExampleFile", () => {
  test("calls writeFile with the correct content", async () => {
    await writeCommandUsageExampleFile(testCommand, commandData);
    globalExpect(writeFile).toHaveBeenCalledWith(
      globalExpect.stringContaining("docs-shopify.dev/commands/examples/topic-test-command.example.sh"),
      "shopify topic test-command [flags]"
    );
  });
});
