// @shopify/cli/dist/hooks/postrun.js
import {
  getNextDeprecationDate
} from "../chunk-KDOLMUJB.js";
import {
  reportAnalyticsEvent
} from "../chunk-Y4AOSTJD.js";
import "../chunk-TMGCRPEZ.js";
import "../chunk-BUQTR47W.js";
import "../chunk-URD52LAO.js";
import "../chunk-2EVDXDJS.js";
import "../chunk-55F2DQCB.js";
import "../chunk-23EONB5V.js";
import {
  addSensitiveMetadata,
  getAllSensitiveMetadata,
  outputDebug,
  renderWarning
} from "../chunk-ZLSD2RNR.js";
import "../chunk-FQIRJADJ.js";
import "../chunk-FBB6KUZG.js";
import "../chunk-OVPEO3GL.js";
import "../chunk-4AOK4O3R.js";
import "../chunk-FWGRGI4R.js";
import "../chunk-BE73NZXB.js";
import "../chunk-VNBOO6TJ.js";
import "../chunk-QTHWTF4X.js";
import "../chunk-RQ24VTW4.js";
import "../chunk-WCNR75S2.js";
import "../chunk-G6FN5VUE.js";
import "../chunk-3JLUTHGR.js";
import "../chunk-GXPKATXW.js";
import {
  init_cjs_shims
} from "../chunk-M63RTPGR.js";

// src/hooks/postrun.ts
init_cjs_shims();

// ../cli-kit/src/public/node/hooks/postrun.ts
init_cjs_shims();

// ../cli-kit/src/public/node/hooks/deprecations.ts
init_cjs_shims();
var postrun = (Command) => {
  const nextDeprecationDate = getNextDeprecationDate();
  if (nextDeprecationDate) {
    const forThemes = Command?.id?.includes("theme");
    renderUpgradeWarning(nextDeprecationDate, forThemes);
  }
};
function renderUpgradeWarning(upgradeByDate, forThemes) {
  const dateFormat = new Intl.DateTimeFormat("default", {
    year: "numeric",
    month: "long",
    day: "numeric"
  });
  const formattedDate = dateFormat.format(upgradeByDate);
  const headline = `Upgrade to the latest CLI version by ${formattedDate}.`;
  const body = "This command requires an upgrade to continue working as intended.";
  const upgradeLink = {
    link: {
      label: "upgrade Shopify CLI",
      url: forThemes ? `https://shopify.dev/docs/themes/tools/cli#upgrade-shopify-cli` : `https://shopify.dev/docs/apps/tools/cli#upgrade-shopify-cli`
    }
  };
  const nextSteps = [["Run", { command: "upgrade" }, "to", upgradeLink]];
  renderWarning({
    headline,
    body,
    nextSteps
  });
}

// ../cli-kit/src/public/node/hooks/postrun.ts
var hook = async ({ config, Command }) => {
  await detectStopCommand(Command);
  await reportAnalyticsEvent({ config, exitMode: "ok" });
  postrun(Command);
  const command = Command?.id?.replace(/:/g, " ");
  outputDebug(`Completed command ${command}`);
};
async function detectStopCommand(commandClass) {
  const currentTime = (/* @__PURE__ */ new Date()).getTime();
  if (commandClass && Object.prototype.hasOwnProperty.call(commandClass, "analyticsStopCommand")) {
    const stopCommand = commandClass.analyticsStopCommand();
    if (stopCommand) {
      const { commandStartOptions } = getAllSensitiveMetadata();
      await addSensitiveMetadata(() => ({
        commandStartOptions: {
          ...commandStartOptions,
          startTime: currentTime,
          startCommand: stopCommand
        }
      }));
    }
  }
}
export {
  hook as default
};
