// @shopify/cli/dist/chunk-T3QDR67D.js
import {
  demo
} from "./chunk-G56QICQP.js";
import {
  base_command_default
} from "./chunk-7VU6FOLD.js";
import {
  outputInfo,
  readFile,
  renderAutocompletePrompt
} from "./chunk-ZLSD2RNR.js";
import {
  joinPath
} from "./chunk-G6FN5VUE.js";
import {
  init_cjs_shims
} from "./chunk-M63RTPGR.js";

// src/cli/commands/demo/catalog.ts
init_cjs_shims();
import { fileURLToPath } from "url";
var Catalog = class extends base_command_default {
  static {
    this.description = "Browse the catalog for steps";
  }
  static {
    this.hidden = true;
  }
  async run() {
    const catalogFile = joinPath(fileURLToPath(import.meta.url), "../../../../../assets/demo-catalog.json");
    const { steps } = JSON.parse(await readFile(catalogFile));
    const stepSelection = await renderAutocompletePrompt({
      message: "Step to display",
      choices: steps.map(({ title, type }) => {
        return {
          label: title,
          value: title,
          group: type
        };
      })
    });
    const selectedStep = steps.find(({ title }) => title === stepSelection);
    outputInfo("The step looks like this:");
    await demo({ steps: [selectedStep] });
    outputInfo("JSON for this step:");
    outputInfo(JSON.stringify(selectedStep, null, 2));
  }
};

export {
  Catalog
};
