// @shopify/cli/dist/custom-oclif-loader-2FLACCMK.js
import {
  fileExistsSync
} from "./chunk-ZLSD2RNR.js";
import "./chunk-FQIRJADJ.js";
import "./chunk-FBB6KUZG.js";
import "./chunk-OVPEO3GL.js";
import {
  require_lib
} from "./chunk-4AOK4O3R.js";
import "./chunk-FWGRGI4R.js";
import "./chunk-BE73NZXB.js";
import "./chunk-VNBOO6TJ.js";
import "./chunk-QTHWTF4X.js";
import "./chunk-RQ24VTW4.js";
import "./chunk-WCNR75S2.js";
import {
  cwd,
  joinPath
} from "./chunk-G6FN5VUE.js";
import "./chunk-3JLUTHGR.js";
import "./chunk-GXPKATXW.js";
import {
  __toESM,
  init_cjs_shims
} from "./chunk-M63RTPGR.js";

// ../cli-kit/src/public/node/custom-oclif-loader.ts
init_cjs_shims();
var import_core = __toESM(require_lib(), 1);
var ShopifyConfig = class extends import_core.Config {
  constructor(options) {
    const path = sniffForPath() ?? cwd();
    if (fileExistsSync(joinPath(`${path}`, "package.json"))) {
      options.pluginAdditions = {
        core: ["@shopify/cli-hydrogen"],
        path
      };
    }
    super(options);
    this["determinePriority"] = this.customPriority;
  }
  customPriority(commands) {
    const oclifPlugins = this.pjson.oclif?.plugins ?? [];
    const commandPlugins = commands.sort((aCommand, bCommand) => {
      const pluginAliasA = aCommand.pluginAlias ?? "A-Cannot-Find-This";
      const pluginAliasB = bCommand.pluginAlias ?? "B-Cannot-Find-This";
      const aIndex = oclifPlugins.indexOf(pluginAliasA);
      const bIndex = oclifPlugins.indexOf(pluginAliasB);
      if (aCommand.pluginType === "core" && bCommand.pluginAlias === "@shopify/cli-hydrogen") {
        return 1;
      }
      if (aCommand.pluginAlias === "@shopify/cli-hydrogen" && bCommand.pluginType === "core") {
        return -1;
      }
      if (aCommand.pluginType === "core" && bCommand.pluginType === "core") {
        return aIndex - bIndex;
      }
      if (bCommand.pluginType === "core" && aCommand.pluginType !== "core") {
        return 1;
      }
      if (aCommand.pluginType === "core" && bCommand.pluginType !== "core") {
        return -1;
      }
      if (aCommand.pluginType === "jit" && bCommand.pluginType !== "jit") {
        return 1;
      }
      if (bCommand.pluginType === "jit" && aCommand.pluginType !== "jit") {
        return -1;
      }
      return 0;
    });
    return commandPlugins[0];
  }
};
function sniffForPath() {
  const pathFlagIndex = process.argv.indexOf("--path");
  if (pathFlagIndex === -1)
    return;
  const pathFlag = process.argv[pathFlagIndex + 1];
  if (!pathFlag || pathFlag.startsWith("-"))
    return;
  return pathFlag;
}
export {
  ShopifyConfig
};
