// @shopify/cli/dist/node-package-manager-HBYFQNJQ.js
import {
  FindUpAndReadPackageJsonNotFoundError,
  PackageJsonNotFoundError,
  UnknownPackageManagerError,
  addNPMDependencies,
  addNPMDependenciesIfNeeded,
  addNPMDependenciesWithoutVersionIfNeeded,
  addResolutionOrOverride,
  bunLockfile,
  checkForNewVersion,
  findUpAndReadPackageJson,
  getDependencies,
  getPackageManager,
  getPackageName,
  getPackageVersion,
  installNPMDependenciesRecursively,
  installNodeModules,
  lockfiles,
  npmLockfile,
  packageManager,
  packageManagerFromUserAgent,
  pnpmLockfile,
  pnpmWorkspaceFile,
  readAndParsePackageJson,
  usesWorkspaces,
  writePackageJSON,
  yarnLockfile
} from "./chunk-55F2DQCB.js";
import "./chunk-23EONB5V.js";
import "./chunk-ZLSD2RNR.js";
import "./chunk-FQIRJADJ.js";
import "./chunk-FBB6KUZG.js";
import "./chunk-OVPEO3GL.js";
import "./chunk-4AOK4O3R.js";
import "./chunk-FWGRGI4R.js";
import "./chunk-BE73NZXB.js";
import "./chunk-VNBOO6TJ.js";
import "./chunk-QTHWTF4X.js";
import "./chunk-RQ24VTW4.js";
import "./chunk-WCNR75S2.js";
import "./chunk-G6FN5VUE.js";
import "./chunk-3JLUTHGR.js";
import "./chunk-GXPKATXW.js";
import "./chunk-M63RTPGR.js";
export {
  FindUpAndReadPackageJsonNotFoundError,
  PackageJsonNotFoundError,
  UnknownPackageManagerError,
  addNPMDependencies,
  addNPMDependenciesIfNeeded,
  addNPMDependenciesWithoutVersionIfNeeded,
  addResolutionOrOverride,
  bunLockfile,
  checkForNewVersion,
  findUpAndReadPackageJson,
  getDependencies,
  getPackageManager,
  getPackageName,
  getPackageVersion,
  installNPMDependenciesRecursively,
  installNodeModules,
  lockfiles,
  npmLockfile,
  packageManager,
  packageManagerFromUserAgent,
  pnpmLockfile,
  pnpmWorkspaceFile,
  readAndParsePackageJson,
  usesWorkspaces,
  writePackageJSON,
  yarnLockfile
};
